### PR TITLE
CI: Fix nightly lint error & fix rust nightly version

### PR DESCRIPTION
**Pull request by Xynnn007**
1. Fixes nightly lint error
2. Follow [the discussion](https://github.com/confidential-containers/confidential-containers/issues/193) from @portersrc , I fix the version of rust nightly compiler in GHA to avoid unexpected CI errors everytime new rust nightly version is released.

### CODE REVIEW

**Comment by mkulke** [comment]
there is a case for testing against nightly: you will pick up changes early that will propagate to stable in later releases, like the last nightly breakage that we've seen.

File: .github/workflows/kbs-rust.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
         rust:
           - stable
           - beta
-          - nightly
+          - nightly-2024-02-18
```
Comment (mkulke) [inline .github/workflows/kbs-rust.yml:28]: why fixing nightly to an (arbitrary?) date, would that give us any additional confidence?

**Comment by portersrc** [comment]
I made a few comments in https://github.com/confidential-containers/confidential-containers/issues/193

I would just drop nightly @mkulke. rust-nightly doesn't necessarily hit stable (if I understand correctly), and even rust-beta is overhead we don't need until we're ready to bump to a stable version.

But FWIW, I'd say @Xynnn007's pinned nightly is better than nightly, b/c of the reasons he stated. Arbitrary date is fine.

**Comment by mkulke** [comment]
> I made a few comments in [confidential-containers/confidential-containers#193](https://github.com/confidential-containers/confidential-containers/issues/193)
> 
> I would just drop nightly @mkulke. rust-nightly doesn't necessarily hit stable (if I understand correctly), and even rust-beta is overhead we don't need until we're ready to bump to a stable version.
> 
> But FWIW, I'd say @Xynnn007's pinned nightly is better than nightly, b/c of the reasons he stated. Arbitrary date is fine.

no, not everything that hits nightly will end up in stable, but more specifically: if something doesn't compile in nightly today it's likely that eventually it will also break in stable, because the original behaviour was based on a bug. not sure if this is a hard rule, but that has been my experience. So in this respect nightly can be a good indicator of upcoming changes that need be dealt with.

that said, I wouldn't mind dropping beta and nightly (I don't see the point in testing against _some_ revision of nightly, but I might be missing something) from the matrix. I don't think the project is affected too much by those changes.

**Comment by Xynnn007** [comment]
@mkulke Right. Nightly test will tell us the potential error asap. however, there might be cases where indirect dependency breaks, which will block the daily development. At the same time, if we update nightly version periodically, we also might need to handle a bunch of errors one time.

This is a every-day-a-little vs a-batch-one-time problem. I am okay with both tbh temporarily as they both has pros and cons. If finally it is decided not to fix nightly version, I will drop the commit in this PR.

**Comment by mkulke** [comment]
yeah, this is mostly bikeshedding, so I while I have opinions, it probably does not matter 😅  but I'm intellectually curious: what good would pinning a nightly version does vs just dropping nightly from the matrix? 

would there be a checkbox item on the release list that says "bump the nightly version and fix all potential errors?"

**Comment by fitzthum** [comment]
Personally I think I lean more towards dropping nightly. We don't make any claims about supporting it. I don't see much benefit in testing with it.

**Comment by Xynnn007** [comment]
good. So it seem clear that we do not need to check against nightly and beta anymore. Let me just delete them in this PR : )

File: .github/workflows/kbs-rust.yml [hunk 2]
```diff
@@ -24,8 +24,6 @@ jobs:
       matrix:
         rust:
           - stable
```
Comment (mythi) [inline .github/workflows/kbs-rust.yml:26]: The original proposal also included that we fix the version and not use a moving target, i.e.,
```suggestion
          - 1.76.0
```

with these updates, it's also possible to drop the `matrix` construct altogether.

**Review by mkulke — APPROVED** [review]

**Comment by portersrc** [comment]
Thanks @Xynnn007 for fielding this. Maybe catch [kbs-e2e](https://github.com/confidential-containers/kbs/blob/main/.github/workflows/kbs-e2e.yaml#L36) and [kbs-docker-e2e](https://github.com/confidential-containers/kbs/blob/main/.github/workflows/kbs-docker-e2e.yaml#L23) in this PR, as well.

**Pull request merged**